### ISSUE TITLE
tutorial

### ISSUE THREAD

**Issue by monkeytypegeorge** [Miodec/monkeytype#617]

**Comment by ThatOneCalculator** [comment]
1. Make a detailed title
2. Make an issue comment
3. Who needs...a tutorial on how to use a keyboard?

**Comment by TheeKevin** [comment]
@ThatOneCalculator this guy is a bot... so everything he says is automated.

Also we need a tutorial on how to use the site. Like some people ask "we need certain symbols for coding!" when you can enable punctuation or do it on quote mode. In which a tutorial would be very nice to tell you about the obscure things on the website that people don't know exist.

**Comment by Miodec** [comment]
Yeah there are many people asking super basic questions in the discord server all the time, like asking how to type quotes, or type more than 30 seconds.

**Comment by TheeKevin** [comment]
@Miodec how would you go about in making a tutorial?

Would it be one of those where there's a text box near a feature explaining it? Or a really long document with pictures and diagrams. Perhaps a giant flow chart?

I figure it'd have to be a tutorial that's good at explaining things while also being flexible (it can be changed or added to easily) plus since there's so much in Monkeytype you'd have to make really specific things in the tutorial easy to find.

**Comment by Miodec** [comment]
The floating text box kind. It highlights different features and explains what they do.

**Comment by RealCyGuy** [comment]
maybe something like the one on https://typing.works

**Comment by TheeKevin** [comment]
Since this issue isn't gaining much traction I would like to propose an idea. As I've been scrolling through Monkeytype I've realized that Monkeytype itself is kind of a tutorial as basically everything explains itself. Instead of a tutorial box thingy how about creating a Q&A that lists basic questions that get answered that people can look at before asking on Discord or Github.

**Comment by DavidPotvin-jpg** [comment]
Hi @TheeKevin this is a great idea and it is a feature I would like to work on !

**Comment by UnrealApex** [comment]
We could have something like VS Code's user interface overview:

![VS Code's user interface overview](https://user-images.githubusercontent.com/79121360/134430898-c89fbfc4-7b5f-4a8f-8d71-cfeb0f224092.png)

**Comment by jacksonwalker0** [comment]
Can I make a pop-up screen that displays an embedded video?

**Comment by Miodec** [comment]
Unassigning due to inactivity

**Comment by anikait17a** [comment]
ok, I'll try my best to do that. Thanks!!

**Comment by anikait17a** [comment]
Sorry, but after thinking a bit about it I don't think that I will be able to do it I'll start with easier ones. sorry

**Comment by ClumsyClover** [comment]
@Miodec, hi I have started working on this issue, I was waiting to get a good starting point to finally ask and get the issue assigned to me, I was getting the dev env setup and generally going through the codebase to understand it. I think I have now gotten to a good point and I would request that this issue be assigned to me.

> Sure, but i think i would like it to be an FAQ window instead, where you have a list of topics on the left, and the right displays the contents. Then add a link to that popup in the footer.
I am currently following the requirements you've mentioned here.

I will further discuss more details about the tutorial popup here as I proceed with the development.

**Comment by aikooo7** [comment]
Hello @ClumsyClover may I ask how is going? Need any help?

**Comment by ClumsyClover** [comment]
@aikooo7 Hey there, I've just been busy with some other things, but I wouldn't mind if you were to have a look at my fork and work on it further or change it however you wished. Also feel free to further discuss things here or discord (@clumsyclover) and I will be happy to chat whenever I can.

**Comment by aikooo7** [comment]
Hello @ClumsyClover I didn't see your ping my bad, I will look at your fork and add you in discord if needed.

**Comment by ClumsyClover** [comment]
Hey @aikooo7, that's okay haha, I was wondering what up with this; but that's completely fine, feel free to DM me on Discord. I would be more than fine to discuss this as I do still want this to be added to moneytype.

**Comment by aikooo7** [comment]
Hello @Miodec , I'm thinking in using markdown for writing tutorials since it's easier to anyone write, I'm also thinking in using [marked](https://github.com/markedjs/marked) for the markdown parsing, what you think about it?

**Comment by Miodec** [comment]
I guess you can use markdown,sure.

**Comment by aikooo7** [comment]
> I guess you can use markdown,sure.

Hey miodec sorry for the dumb question but I want to read a markdown file that will be in static/markdown but I can't run fs module inside popous, should I create a function in backend/src/utils/filesystem?

**Comment by Miodec** [comment]
You can base it off of the json file loading in misc.ts.

**Comment by NullyIsHere** [comment]
I found this some days ago that might help. https://github.com/bluewave-labs/guidefox

**Comment by NullyIsHere** [comment]
Did notice its two years old but i might be able to work on it if needed :D

**Comment by openvaibhav** [comment]
Hey @Miodec I was working on this for sometime and came up with something, Can you please assign this to me.

I've implemented this as a FAQ modal as suggested. It includes:

A list of topics on the left panel
Content displayed on the right panel
A live search box that filters topics as you type
A link in the footer to open it
[More FAQs can be added]

PR: #7601
